### PR TITLE
[FLINK-1875] Add figure explaining slots and parallelism

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

**Comment by mxm** [comment]
That's a very nice graphical explanation of parallelism. :+1: 

You could add another row with a job which changes parallelism in the middle of the job. For example, caused by a GroupReduce without a grouping. That would give users a hint about the kind of bottleneck such a transformation causes.

**Comment by rmetzger** [comment]
I would like to add a second picture that shows how the data is streamed between the operators for the parallelism=1 sink.
We could add a all reduce as a variant to that.

**Comment by rmetzger** [comment]
It seems that the apache rat plugin has troubles verifying the license header of the svg file.

I have to look into this again.

**Comment by rmetzger** [comment]
The issue seems to be fixed

File: docs/config.md [hunk 1]
```diff
@@ -370,8 +370,7 @@ system, such as */tmp* in Linux systems.
 
 ### Configuring TaskManager processing slots
 
-A processing slot allows Flink to execute a distributed DataSet transformation, such as a
-data source or a map-transformation.
+A processing slot allows Flink to execute an instance of a job.
```
Comment (fhueske) [inline docs/config.md]: I think the term *instance of a job* might be misleading. The document uses *operator instance* or *(user) function instance* but IMO it is not clear what an *instance of a job* is.

I agree that the previous description wasn't good either, but we might try to come up with something better.
What about "Flink executes a program in parallel by splitting it into subtasks and scheduling these subtasks to processing slots."?

**Comment by fhueske** [comment]
Nice figure! I added a comment about a certain term in the documentation, but otherwise I think we should get it in.

File: docs/config.md (hunk 1 above)
Comment (rmetzger) [inline docs/config.md]: I updated the text with your suggestion.

**Pull request merged**